client/client.go

```diff
@@ -454,7 +454,9 @@ func (c *Client) handleGreetAndStartReading() error {
 
 	// Make sure to start reading after we have set up this handler, otherwise
 	// some messages will be lost.
-	go c.read(greeted)
+	go func() {
+		done <- c.read(greeted)
```

Comment:
We probably need to create the `done` channel with a buffer of 2, in case both this goroutine and the other one fails. Otherwise we might end up in a goroutine leak.